array.go

```diff
@@ -67,6 +67,44 @@ func (a *Array) Raw() []interface{} {
 	return a.value
 }
 
+// Decode unmarshals the underlying value attached to the Array to a target variable.
+// target should be one of these:
+//
+// 1) pointer to empty interface
+//
+// 2) pointer to slice of empty interfaces
+//
+// 3) pointer to slice of structs
```

Comment:
Actually this list is is incomplete. It can be slice of any type into which array values can be unmrashalled. E.g. https://go.dev/play/p/p64my6HNdXi

Reply:
Yes, that makes sense. So now, do I just add "pointer to slice of any type"?

Reply:
Also, could please change list formatting a bit, here and in other methods? To match the rest of the lib.

```
// target should be one of these:
//  - pointer to an empty interface
//  - pointer to a slice of any type
```